sql/core/src/main/scala/org/apache/spark/sql/jdbc/H2Dialect.scala

```diff
@@ -41,6 +43,45 @@ private[sql] object H2Dialect extends JdbcDialect {
   override def isSupportedFunction(funcName: String): Boolean =
     supportedFunctions.contains(funcName)
 
+  class H2SQLBuilder extends JDBCSQLBuilder {
+    override def visitUserDefinedScalarFunction(
+        funcName: String, canonicalName: String, inputs: Array[String]): String = {
+      canonicalName match {
+        case "H2.CHAR_LENGTH" =>
```

Comment:
I don't think this makes sense. In production, the H2 dialect does not expose the `H2.CHAR_LENGTH` UDF. How can it work in a non-test environment?